internal/infrastructure/kubernetes/applier/apply.go

```diff
@@ -122,7 +124,9 @@ func (i *Instance) CreateOrUpdateService(ctx context.Context, svc *corev1.Servic
 		}
 	} else {
 		// Update if current value is different.
-		if !reflect.DeepEqual(svc.Spec, current.Spec) {
+		// Only compare the selector and ports(not include nodePort) in case user have modified for some scene.
+		if !cmp.Equal(svc.Spec.Selector, current.Spec.Selector) &&
```

Comment:
can you move these to a single function for better extension in the future?

Reply:
ok